Exp-001 - algo, UI, Gem

solminwoo opened the pull request:
UI - added a EXP UI to keep track of exp and levels.

Gem - Created Blue Green Red ExpGems

Algo - Gain the jem if the player get closer

solminwoo committed:
Exp-001 - algo, UI, Gem

UI - added a EXP UI to keep track of exp and levels.

Gem - Created Blue Green Red ExpGems

Algo - Gain the jem if the player get closer

allenkim0129 commented on Assets/Scripts/Exp.cs:
```diff
@@ -0,0 +1,44 @@
+
+    private void Update()
+    {
+        if (currentExp / maxExp >= 1)
```
Avoid using division if possible. Division is an expensive operation.

if (currentExp - maxExp >= 0)

allenkim0129 commented on Assets/Scripts/Exp.cs:
```diff
@@ -0,0 +1,44 @@
+        }
+    }
+
+    public void gainExp(int x)
```
why using int type? Let's use the same type, float, as currentExp and maxExp.

allenkim0129 commented on Assets/Scripts/ExpGem/ExpGem.cs:
```diff
@@ -0,0 +1,49 @@
+
+    private void Update()
+    {
+        if(getDistanceToPlayer() < gainRadius)
```
Let's put gainRadius in Player, since this gainRadius can grow as players increase the radius

player.gainRadius

allenkim0129 commented on Assets/Scripts/ExpGem/ExpGem.cs:
```diff
@@ -0,0 +1,49 @@
+        if(getDistanceToPlayer() < .1f)
+        {
+            Destroy(gameObject);
+            exp.gainExp(expAmount);
```
Shouldn't we destroy gameObject after gainExp()?

allenkim0129 commented on Assets/Scripts/ExpGem/ExpGem.cs:
```diff
@@ -0,0 +1,49 @@
+    public Exp exp;
+    public Player player;
+
+    public int expAmount;
```
Change to float.

solminwoo committed:
Code Review Fix

Changed int -> float
moved gainRadius to player and delete some code accordingly.
division to substraction
